dev/integration_tests/web_e2e_tests/test_driver/text_editing_integration.dart

```diff
@@ -0,0 +1,275 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:html';
+import 'dart:js_util' as js_util;
+import 'package:flutter/gestures.dart';
+import 'package:flutter/services.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:web_e2e_tests/text_editing_main.dart' as app;
+import 'package:flutter/material.dart';
+
+import 'package:integration_test/integration_test.dart';
+
+void main() {
+  IntegrationTestWidgetsFlutterBinding.ensureInitialized();
+
+  testWidgets('Focused text field creates a native input element',
```

Comment:
Wow, this test is involved, and text edition is still changing quite a bit, do you think this can break because of legit changes in the way text editing is handled in the engine? Can this prevent engine rolls to flow into flutter?

Reply:
Thanks! yes this is a very good test.

I think the current situation is worse (which happened a few times)
- If there is a change on the framework, engine tests won't run and won't break. 
- developer will merge the change to flutter/flutter
- engine developers will start seeing failure on random PRs for felt firefox step (since it's the first step that runs integration tests)
- Later (after a research probably) the flutter/flutter PR will be reverted
- PR author will try to test the PR running felt locally, this is also a little tricky since there are many flutter/flutter developers who does not normally do engine development

In the new process,
- flutter/flutter developers would be protected against the changes to e2e and also against platform channels and text editing changes
- flutter/engine developers, will break the the tests on the engine side first if they are making a change that would change the communication between flutter and the web engine (so the tree will be read)
- If the change is on purpose, then there should be a handshake (there should be 3 PRs in a row, first one would change the test or the code to a temporary state. Last one will fix it.)

I think this is good since this is also reminds both engine/flutter PRs about the other repo. If this was a package (rather than the engine itself) best could have been to use a version in the pubspec of course. But I think we have this problem for many issues between engine<->flutter repo.

Thanks for the reminder though I'm chat with @Hixie to see what can we do for this type of changes. May be we can add a guide for developers.